app/utils/use-tab-visibility.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useEffect, useMemo,useState } from 'react';
```

Comment:
Why did this file have to get brought in? Is this needed to improve storybook / tests?

Reply:
I did this because `use-tab-visibility` is built incorrectly. It lacks dist for es-modules: https://publint.dev/use-tab-visibility@1.0.9.
That raises issues with adding components using that hook into stories. I want to make a PR to it initially, but repo as archived.

Reply:
Thank you!